File: 189/c.rb
# TLE-
n = gets.to_i
a = gets.chomp.split.map(&:to_i)

ans = 0
(n).times do |l|
  x = a[l]
  l.upto(n-1) do |r|
    x = [x,a[r]].min
    ans = [ans, (r-l+1) * x].max
  end
end
puts ans

File: 189/c.go
package main

import (
	"fmt"
)

func main() {
	var n int
	fmt.Scan(&n)

	a := make([]int, 0)
	for i := 0; i < n; i++ {
		var num int
		fmt.Scan(&num)
		a = append(a, num)
	}

	ans := 0
	for i := 0; i < n; i++ {
		x := a[i]
		for j := i; j < n; j++ {
			x = min(x, a[j])
			ans = max(ans, x*(j-i+1))
		}
	}
	fmt.Println(ans)
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func min(a, b int) int {
	if a > b {
		return b
	}
	return a
}
